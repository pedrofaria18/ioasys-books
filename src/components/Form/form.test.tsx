import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { Form } from '.'

describe('Form', () => {
  it('should render correctly', () => {
    const { container } = render(<Form />)

    expect(container).toMatchSnapshot()
  })

  it('should render the email and password input', async () => {
    render(<Form />)

    const emailInput = await screen.findByLabelText('Email')
    const passwordInput = await screen.findByLabelText('Senha')

    expect(emailInput).toBeInTheDocument()
    expect(passwordInput).toBeInTheDocument()

    expect(emailInput).toHaveAttribute('type', 'email')
    expect(passwordInput).toHaveAttribute('type', 'password')

    userEvent.type(emailInput, 'desafio@ioasys.com.br')
    userEvent.type(passwordInput, '12341234')

    expect(emailInput).toHaveValue('desafio@ioasys.com.br')
    expect(passwordInput).toHaveValue('12341234')
  })

  it('should render the submit button', async () => {
    render(<Form />)

    const submitButton = await screen.findByText('Entrar')

    expect(submitButton).toBeInTheDocument()
    expect(submitButton).toHaveAttribute('type', 'submit')
  })
})
