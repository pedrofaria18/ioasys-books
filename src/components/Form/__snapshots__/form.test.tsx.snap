// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render correctly 1`] = `
<div>
  <div
    class="sc-eCYdqJ hQDWrj"
  >
    <div
      class="sc-gsnTZi kchGOK"
    >
      <div
        class="sc-dkzDqf eIgzCe"
      >
        <label
          for="email"
        >
          Email
        </label>
        <input
          id="email"
          name="email"
          type="email"
          value=""
        />
      </div>
    </div>
    <div
      class="sc-gsnTZi kchGOK"
    >
      <div
        class="sc-dkzDqf eIgzCe"
      >
        <label
          for="password"
        >
          Senha
        </label>
        <input
          id="password"
          name="password"
          type="password"
          value=""
        />
      </div>
      <button
        type="submit"
      >
        Entrar
      </button>
    </div>
  </div>
</div>
`;
